### ISSUE TITLE
Change in GaussianProcessRegressor behaviour causes issues with uncertainty prediction

### ISSUE THREAD

**Issue by LiamWilbraham** [brianhie/uncertainty#3]
Hi there, 

Congrats on the great work!

When trying to reproduce some results from your paper, I ended up with slightly different Kd mean values and very different (orders of magnitude higher) values for the standard deviation computed by the scikit-learn gp backend. Specifically, I am doing:

```
python bin/predict_davis2011kinase.py gp exploit 10 pknb > predict.log 2>&1
```

From which I get, for instance:

```
Acquire (72, 334) ZINC000063539888 <--> PKNB(M.tuberculosis) with predicted Kd value 9823.638 and uncertainty 7126131550.254
```

I managed to track this down to some updated behaviour of the `GaussianProcessRegressor` from scikit-learn, where the behaviour of specifying the `normalise_y` argument has changed. Using the current requirements file by default installs the latest version of scikit-learn with this new  behaviour.  

Explicitly falling back to scikit-learn version 0.22 fixes this issue, and I get the results as shown in the paper, so I think this should be specified in the requirements file, or the code updated to reflect the new behaviour. 

The warning is visible here:
https://scikit-learn.org/stable/modules/generated/sklearn.gaussian_process.GaussianProcessRegressor.html#sklearn.gaussian_process.GaussianProcessRegressor.predict

Thanks a lot,
Liam

**Comment by brianhie** [comment]
@LiamWilbraham wow, thanks for letting me know and a special thanks for tracking down the problem! Added scikit-learn to requirements.txt and updated the README with a troubleshooting section describing this potential problem: https://github.com/brianhie/uncertainty/commit/2a9dd1966632abefa94ecf4c5cf33020cca9d903
